packages/react/tooltip/src/Tooltip.tsx

```diff
@@ -133,10 +134,9 @@ const Tooltip: React.FC<TooltipProps> = (props: ScopedProps<TooltipProps>) => {
     () => stateMachine.send({ type: 'OPEN', id: contentId, delayDuration }),
     [contentId, delayDuration]
   );
-  const handleClose = React.useCallback(
-    () => stateMachine.send({ type: 'CLOSE', id: contentId, skipDelayDuration }),
-    [skipDelayDuration, contentId]
-  );
+  const handleClose = useCallbackRef(() => {
+    stateMachine.send({ type: 'CLOSE', id: contentId, skipDelayDuration });
+  });
 
   // send transition if the component unmounts
   React.useEffect(() => () => handleClose(), [handleClose]);
```

Comment:
The `handleClose` had a `contentId` dependency which would start as `undefined` so this hook would call its cleanup when the `contentId` was set on client and instantly close the `defaultOpen` tooltips.

This fix solves it for now but I need to create a new `TooltipProvider` as part of this React 18 piece to ensure our state is all managed through React state / Context, so I'll see if I can make any improvements here then as well.

Reply:
Nice catch!

Reply:
Took me a while 😅